bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -0,0 +1,265 @@
+ * 
+ * @implNote On activation, a thread is spawned which handles the detection. Scan runs once,
+ *           no continuous background scanning.
+ *
+ * @author Holger Friedrich - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonFinder.class, name = IpAddonFinder.SERVICE_NAME)
+public class IpAddonFinder extends BaseAddonFinder {
+
+    public static final String SERVICE_TYPE = SERVICE_TYPE_IP;
+    public static final String SERVICE_NAME = SERVICE_NAME_IP;
+
+    private final Logger logger = LoggerFactory.getLogger(IpAddonFinder.class);
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_NAME);
+    private @Nullable Future<?> scanJob = null;
+    Set<AddonInfo> suggestions = new HashSet<>();
+
+    public IpAddonFinder() {
+        logger.warn("IpAddonFinder::IpAddonFinder");
+    }
+
+    @Deactivate
+    public void deactivate() {
+        logger.warn("IpAddonFinder::deactivate");
+        stopScan();
+    }
+
+    public void setAddonCandidates(List<AddonInfo> candidates) {
+        logger.warn("IpAddonFinder::setAddonCandidates({})", candidates.size());
+        super.setAddonCandidates(candidates);
+        startScan();
+    }
+
+    synchronized void startScan() {
+        if (scanJob == null) {
+            scanJob = scheduler.schedule(this::scan, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    void stopScan() {
+        Future<?> tmpScanJob = scanJob;
+        if (tmpScanJob != null) {
+            if (!tmpScanJob.isDone()) {
+                logger.trace("Trying to cancel IP scan");
+                tmpScanJob.cancel(true);
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ignore) {
+                }
+            }
+            scanJob = null;
+        }
+    }
+
+    void scan() {
+        logger.warn("IpAddonFinder::scan");
+        for (AddonInfo candidate : addonCandidates) {
+            for (AddonDiscoveryMethod method : candidate.getDiscoveryMethods().stream()
+                    .filter(method -> SERVICE_TYPE.equals(method.getServiceType())).toList()) {
+
+                logger.trace("Checking candidate: {}", candidate.getUID());
+
+                Map<String, String> parameters = method.getParameters().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getValue()));
+                Map<String, String> matchProperties = method.getMatchProperties().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getRegex()));
+
+                // parse standard set op parameters:
+                String type = Objects.toString(parameters.get("type"), "");
+                String request = Objects.toString(parameters.get("request"), "");
+                String response = Objects.toString(matchProperties.get("response"), "");
+                int timeoutMs = 0;
```

Comment:
Maybe use a meaningful default for this if it is not provided, rather than not trying. Define it in a constant.

Reply:
no default needed, just to aviod null for latet parsing

Reply:
String constants are used in the current code.
Unknown type is handled in the code below, defaulting to "" is fine. Empty request frame is a valid use case. 
I did not want to check for all possible parameters, as other types than ipMulticast might not use all of them.